src/tpu_manager.rs

```diff
@@ -94,15 +94,15 @@ impl TpuManager {
             )
             .await?;
             self.error_count.store(0, Ordering::Relaxed);
-            *self.tpu_client.write().await = Arc::new(tpu_client);
+            *self.tpu_client.write().await = tpu_client;
             info!("TPU Reset after 5 errors");
         }
 
         Ok(())
     }
 
-    async fn get_tpu_client(&self) -> Arc<QuicTpuClient> {
-        self.tpu_client.read().await.clone()
+    pub async fn get_tpu_client(&self) -> RwLockReadGuard<QuicTpuClient> {
```

Comment:
Should never ever expose guards outside of the class. I think I had explicitly told you so.